pkg/kubectl/cmd/cp.go

```diff
@@ -162,13 +162,13 @@ func checkDestinationIsDir(dest fileSpec, f cmdutil.Factory, cmd *cobra.Command)
 }
 
 func copyToPod(f cmdutil.Factory, cmd *cobra.Command, stdout, stderr io.Writer, src, dest fileSpec) error {
-	if len(src.File) == 0 {
+	if len(src.File) == 0 || len(dest.File) == 0 {
 		return errFileCannotBeEmpty
 	}
 	reader, writer := io.Pipe()
 
 	// strip trailing slash (if any)
-	if strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/") {
+	if len(dest.File) > 1 && strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/") {
```

Comment:
`len(dest.File) > 0`?

Reply:
@juanvallejo When dest.File == "/" , it will remove suffix '/'  of dest.File which makes dest.File to be empty.
It is the source of bug.
So what I mean here is, when len(dest.File) == 1 , there's no need to remove its suffix '/' at all.

Reply:
Doesn't `strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/")` already take care of dest.File == "/"?